contracts/Campaign.sol

```diff
@@ -0,0 +1,123 @@
+pragma solidity ^0.5.17;
+
+import "./IAM.sol";
+import "./CampaignFactory.sol";
+
+contract Campaign {
+    address manager;        // the factory
+    address owner;
+    uint256 lockoutEndDatetime;
+    IAM IAMContract;
+    uint256 totalDonated = 0;
+    uint256 commissionBP = 1000; // 10% == (1000 / 10,000) basis points
+    uint256 basispoints = 10000;
+
+    event donationMade (address donor, uint256 donatedAmt);
+    event hasWithdrawn (address from, uint256 totalDonationAmt);
+    event hasRefunded (address donor, uint256 refundedAmt);
+
+    constructor(uint256 secs, address orgAddress, IAM IAMaddress 
+        ) public {
+        lockoutEndDatetime = block.timestamp + secs;
+        owner = orgAddress;
+        manager = msg.sender;
+        IAMContract = IAMaddress;
+    }
+
+    // --- MODIFIERS ---
+    modifier ownerOnly() {
+        require(owner == msg.sender);
+        _;
+    }
+
+    modifier verifiedOnly() {
+        require(isVerifiedOwner() == true);
+        _;
+    }
+
+    modifier distrustOnly() {
+        require(isDistrustedOwner() == true);
+        _;
+    }
+
+    modifier managerOnly() {
+        require(manager == msg.sender);
+        _;
+    }
+
+    modifier withinLockoutOnly() {
```

Comment:
Possibly confusing name if Prof sees this, what about something like
1. ongoingCampaignOnly
2. validCampaignOnly

Just some suggestions

Reply:
Updated to ongoingCampaignOnly